index.js

```diff
@@ -244,6 +246,83 @@ class GithubScm extends Scm {
             }
         };
     }
+
+    /**
+     * Get data for a specific repo
+     * @method getCommitSha
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUrl     The scmUrl to get data of
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getRepoInfo(scmInfo, token) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'get',
+                token,
+                params: scmInfo
+            }, (error, repoInfo) => {
+                if (error) {
+                    return reject(error);
+                }
+
+                return resolve(repoInfo);
+            });
+        });
+    }
+
+    /**
+     * Get a url for a specific branch
+     * @method getRepoId
+     * @param  {Object}    config        Configuration
+     * @param  {String}    config.scmUrl The scmUrl to get url of
+     * @param  {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getBranchUrl(scmInfo, token) {
+        return new Promise((resolve, reject) => {
+            this.breaker.runCommand({
+                action: 'getBranch',
+                token,
+                params: scmInfo
+            }, (error, branchInfo) => {
+                if (error) {
+                    return reject(error);
+                }
+
+                // eslint-disable-next-line no-underscore-dangle
+                return resolve(branchInfo._links.html);
+            });
+        });
+    }
+
+    /**
+     * Get repository object for storing new pipeline
+     * @method getRepoId
+     * @param {Object}    config        Configuration
+     * @param {String}    config.scmUrl The scmUrl to generate ID
+     * @param {String}    config.token  The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    _getRepoId(config) {
```

Comment:
It’s my mistake. The function comment in the scm-base should be like `Return an object that includes a unique identifier for the scmUrl, a display name and a link to a repository`.  I’m going to fix it.

I don't have any idea for new natural function name because I'm not good at English. Would you please suggest new one?
@stjohnjohnson said `getRepoId` should return this structure (screwdriver-cd/screwdriver#160), so we should return an object which includes name and url.